### ISSUE TITLE
Players crashing when getting off the bus

### ISSUE THREAD

**Issue by mortdudley** [propstg/esx-blarglebus#29]
Hi, we have an issue where drivers will crash out of the game every time they exit the bus during a run. 

This happens only when peds are set to show on the client. When they are set to show on the server, drivers can exit the bus without issue. I've noticed the script will cause quite high hitch warnings when set to show on server, so we would really like to be able to just show it on client without any issues.

Any insight you might have would be fantastic :)

**Comment by propstg** [comment]
Hi, thanks for the report. I'll try to take a look at it, this evening. Do you know if there are any error messages in the client's log file?

**Comment by mortdudley** [comment]
Thanks for the quick reply. I'll double check and get back to you.

**Comment by mortdudley** [comment]
I've just checked and I can't see anything of interest in the log. Would you like me to send you a copy?

**Comment by propstg** [comment]
Thanks for checking the log. Unfortunately, I wasn't able to recreate the issue. :( Could I please get the client's log after the game crashes, but before relaunching the game, just in case there's some clue hidden in there?

Also, if you wouldn't mind...
1. What server version are you using?
2. Is there an error message displayed to the player when the game crashes?
3. Could I get a copy of your config.lua file, in case it's a combination of settings causing it?
4. Does it only happen occasionally? or every time?
5. Is there a particular route that it happens on? or is it all routes?
6. Are you using a custom route/bus?

Thanks!

**Comment by propstg** [comment]
Hey, @mortdudley, just wondering if you're still experiencing the issue.

**Comment by mortdudley** [comment]
Yeah mate, still got the problem. Sorry I haven’t replied to your last comment, real life got in the way. I’ll get it all the info you requested shortly. Thanks again.

**Comment by propstg** [comment]
No worries! And no rush. Hope everything is well.

**Comment by Tinky124** [comment]
I had that problem on onesync and i back to 32 slots and no problems now..

**Comment by propstg** [comment]
Hey @Tinky124, thanks for the information. I've never looked into onesync, so I have a couple dumb questions, if you don't mind... 

Do you know if setting `+set onesync_enabled 1` with 32 slots actually runs onesync enough to be able to test this? Or do you actually need to set the slots to something greater than 32 (and I guess donate to the patreon?) for the error to occur?

**Comment by Tinky124** [comment]
yes you can with 32 slots test it.. dont need to run more than 32 slots.

Just put +set onesync_enabled 1 in your server.cfg

Also check my pull request i updated to lastest now manifest version..

**Comment by propstg** [comment]
Thanks for the info! I wasn't able to recreate the issue when using onesync--I did a full metro route and got out at almost every stop (and between a few stops) without any issues.

Do you remember if there was an error message when the client crashed? Or are there specific steps that need to be followed for it to happen?

**Comment by mortdudley** [comment]
Hi mate,

Server version is - FXServer-master SERVER v1.0.0.1972 win32

Error Message:
![image](https://user-images.githubusercontent.com/57285895/81889242-a199b800-95d5-11ea-9aaf-6d0102098052.png)

Config file: 
```Config = {}
Config.Locale = 'en'
Config.DebugLog = false

-- Fuel options: These just set the initial fuel to 100%, if you're using the associated fuel script. Be sure to set any others to false.
-- Note:  If using newer versions of LegacyFuel, make sure LegacyFuelFolderName matches the name of the LegacyFuel resource folder, in case you renamed it.
Config.UseLegacyFuel = false
Config.LegacyFuelFolderName = 'LegacyFuel'
Config.UseFrFuel = false

Config.PutPlayerInBusOnSpawn = true
Config.OnlyShowPedsOnClient = true

Config.EnterVehicleTimeout = 10000
Config.DelayBetweenChanges = 1000
Config.DeleteDistance = 100.0
Config.Markers = {
    Size = 10.0,
    StartColor = {r = 20, g = 200, b = 20, a = 100},
    AbortColor = {r = 200, g = 20, b = 20, a = 100},
}

Config.ShowOverlay = true

Config.Routes = {
    AirportRoute,
    ScenicRoute,
    MetroRoute
}

Config.PedModels = {
    'u_m_m_partytarget',
    'u_m_m_prolsec_01',
    'u_m_m_promourn_01',
    'u_m_m_rivalpap',
    'u_m_m_spyactor',
    'u_m_m_willyfist',
    'u_m_o_finguru_01',
    'u_m_o_taphillbilly',
    'u_m_o_tramp_01',
    'u_m_y_abner',
    'u_m_y_antonb',
    'u_m_y_babyd',
    'u_m_y_baygor',
    'u_m_y_burgerdrug_01',
    'u_m_y_chip',
    'u_m_y_cyclist_01',
    'u_m_y_fibmugger_01',
    'u_m_y_guido_01',
    'u_m_y_gunvend_01',
    'u_m_y_hippie_01',
    'u_m_y_imporage',
    'u_m_y_mani',
    'u_m_y_militarybum',
    'u_m_y_paparazzi',
    'u_m_y_party_01',
    'u_m_y_pogo_01',
    'u_m_y_proldriver_01',
    'u_m_y_rsranger_01',
    'u_m_y_staggrm_01',
    'u_m_y_tattoo_01',
    'u_m_y_zombie_01'
}
```
It appears to happen every time you get off the bus while passengers are either entering or exiting the bus and only when shown on the client only. It also appears to happen on all routes. We don't use any custom routes or buses. I also tested it on our test server that does not have Onesync enabled and we don't have the issue at all there.

I have a copy of the crash dump also if you would like to see it.

**Comment by propstg** [comment]
Awesome, thank you @mortdudley! The only thing that really stands out is that your server is win32, and I've only ever tested on a Linux server. I'll try setting up/testing on a Windows server, maybe sometime this weekend.

@Tinky124, are you running your server on Windows, as well?

**Comment by Tinky124** [comment]
I am running also server on Windows.. @propstg Lastest Fivem Server Artifact.

**Comment by propstg** [comment]
@Tinky124 Hmm, does the version reported by the server say win64? I installed it on a 64 bit machine and it says `win32`, here. Do you happen to know what server version you were running when you saw the issue on onesync, before?

Potentially important, I did stupidly grab the latest server version when I set up my server, last night. It looks like the earliest version on the artifact server is 2000, and I can't get that to work with the latest es_extended (the error _appears_ to be because it's expecting a __resource.lua file and es_extended is now using an fxmanifest file, but I'm not 100% sure if that's the cause). 

(Also potentially important, it looks like the latest version of ESX removed ES as a dependency, which explains why the installation instructions were much simpler than I remembered and also means I probably won't try downgrading ESX on my windows server to see if the issue with this script is just magically fixed in newer server versions until tomorrow night at the earliest, because 🍻)

**Comment by Tinky124** [comment]
@propstg maybe beacuse of this commit #12 ?

**Comment by propstg** [comment]
@Tinky124,

I mean, it is possible, but I'd be surprised if that specific change is the cause. It doesn't look like there are any natives used in that change that weren't already being used elsewhere in the script.

If you think it is the cause, and you're still able to recreate the issue with onesync enabled, you could try commenting these lines out of `startMainLoop`?
```
            elseif isPlayerNotInBus() then
                ESX.ShowHelpNotification(_('get_back_in_bus'))
                Citizen.Wait(5)
```

**Comment by Tinky124** [comment]
@propstg i tested and its not problem with that, peds.lua maybe because when peds are in bus and when you leave the bus then crash happen..

**Comment by propstg** [comment]
Per #32, it sounds like disabling ped spawning _might_ be a workaround for the issue (or some not-fully-stated reason). I've added config options to make this easy.

I'll keep this open, even if that workaround works, because driving an empty bus around is silly.

**Comment by EGUltraTM** [comment]
@propstg 
Hope you can make it like the ambulance missions to fix it. 

for me npcs spawn on first locations but then they dont spawn anymore and i drive an empty bus around.

**Comment by propstg** [comment]
Hi EGUltraTM,

> Can you maybe fix the npcs.
Like when they go out of the bus the walk away?

That sounds like a good idea, but might be better-suited for a new issue, if you'd like to open one for it.

> for me npcs spawn on first locations but then they dont spawn anymore and i drive an empty bus around.

Also, I saw this in my email, but it looks like you edited it out. Is this still needed? I'm not sure if it'll fix the issues on this ticket, but it seems like it couldn't hurt to use the same style of ped spawning as the ambulance mission script.

Thanks!

**Comment by EGUltraTM** [comment]
Looks like it was a onetime issue where no npcs spawned. 

ok will create a issue with the npc thing. 

it looks wierd if all try to run to one point :D
